### ISSUE TITLE
[Python] `XGBClassifier.predict_proba` returns incorrect first column for objective `binary:logitraw`

### ISSUE THREAD

**Issue by david-cortes** [dmlc/xgboost#11373]
If one uses objective `binary:logitraw` with the scikit-learn classifier class, method `predict_proba` will return something incorrect for the first column. I think it is incorrectly adding the `base_score` to that column where it shouldn't, but not 100% sure.

Example:
```python
import numpy as np
import xgboost as xgb
rng = np.random.default_rng(seed=123)
X = rng.standard_normal(size=(100,4))
y = rng.integers(2, size=X.shape[0])
model = xgb.XGBClassifier(objective="binary:logitraw").fit(X,y)
model.predict_proba(X[:5])
```
```
array([[-1.0884051,  2.0884051],
       [-1.071372 ,  2.071372 ],
       [-4.0730515,  5.0730515],
       [-2.1187818,  3.1187818],
       [-2.2682145,  3.2682145]], dtype=float32)
```

Problem: if one applies a sigmoid transformation to these numbers, the obtained probabilities do not sum to 1:
```python
model.predict_proba(X[:5])
1 / (1 + np.exp(-model.predict_proba(X[:5])))
```
```
array([[0.25191873, 0.8897711 ],
       [0.25514224, 0.88808936],
       [0.01674035, 0.9937757 ],
       [0.10728469, 0.95766085],
       [0.09378985, 0.9633221 ]], dtype=float32)
```

... and if we apply a softmax transformation instead, they don't match with the probabilities that one would obtain with raw values:
```python
from scipy.special import softmax
softmax(
    model.predict_proba(X[:5]),
    axis=1
)
```
```
array([[4.00477834e-02, 9.59952235e-01],
       [4.13781367e-02, 9.58621919e-01],
       [1.06623185e-04, 9.99893308e-01],
       [5.28510520e-03, 9.94714916e-01],
       [3.92510649e-03, 9.96074855e-01]], dtype=float32)
```

```python
raw = model.get_booster().inplace_predict(X[:5])
1 / (1 + np.exp(-raw))
```
```
array([0.8897711 , 0.88808936, 0.9937757 , 0.95766085, 0.9633221 ],
      dtype=float32)
```

From what I can tell, the second column from `predict_proba` has correct numbers, but no the first one.

**Comment by trivialfis** [comment]
The logitraw is more a legacy. One can achieve the same with `output_margin`. I will look into this later.

**Comment by EthicalCha0s** [comment]
It looks like the root cause is that `binary:logitraw` emits raw margins, but `predict_proba` feeds them straight into the 2-column expansion as if they were already probabilities, so the first column is `1 - raw` and ends up outside `[0, 1]`.

Applying a sigmoid to the raw logits inside `predict_proba` could make it return valid probabilities (consistent with `binary:logistic`).

**Comment by RAMitchell** [comment]
Isn't binary:logitraw redundant? You can just use the normal binary objective and ask for margin output. In which case we should deprecate/remove.

**Comment by EthicalCha0s** [comment]
Agreed, that makes sense. I've closed my PR (#12272) in favour of that direction. Happy to help with the deprecation if it'd be useful.
